modules/veteran/app/controllers/veteran/v0/vso_accredited_representatives_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Veteran
+  module V0
+    class VSOAccreditedRepresentativesController < BaseAccreditedRepresentativesController
+      before_action :verify_type
+
+      PERMITTED_TYPE = 'veteran_service_officer'
+
+      private
+
+      def serializer_class
+        'Veteran::Accreditation::VSORepresentativeSerializer'.constantize
+      end
+
+      def representative_query
+        query = base_query
+                .joins('JOIN LATERAL UNNEST(veteran_representatives.poa_codes) AS UnnestedPoaCode ON true')
+                .joins('JOIN veteran_organizations ON UnnestedPoaCode = veteran_organizations.poa')
+                .select("veteran_representatives.*, array_agg(veteran_organizations.name ORDER BY veteran_organizations.name) AS organization_names, #{distance_query_string}") # rubocop:disable Layout/LineLength
+                .where('? = ANY(veteran_representatives.user_types)', search_params[:type])
+                .group(Veteran::Service::Representative.column_names.map { |col| "veteran_representatives.#{col}" })
```

Comment:
This group by could make the query pretty inefficient. We should keep an eye on response times to see if this needs optimized.

Reply:
Agreed we'll address this when we move the code into our own module and can create our own tables